api/app/routers/pteams.py

```diff
@@ -509,11 +509,50 @@ def set_ticket_status(
     if data.topic_status == models.TopicStatusType.alerted:
         # user cannot set alerted
         raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Wrong topic status")
-    if data.topic_status == models.TopicStatusType.scheduled and not data.scheduled_at:
+    if data.topic_status == models.TopicStatusType.scheduled and data.scheduled_at is None:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail="If statsu is schduled, specify schduled_at",
         )
+
+    if _current := ticket.current_ticket_status.ticket_status:
+        prev_topic_status = _current.topic_status
+    else:
+        prev_topic_status = models.TopicStatusType.alerted
+
+    if (
+        prev_topic_status == models.TopicStatusType.scheduled
+        and data.topic_status is None
+        and data.scheduled_at is None
+    ):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="If statsu and scheduled_at is None, put values in status and scheduled_at",
+        )
+
+    if (
+        data.topic_status != models.TopicStatusType.scheduled
+        and data.scheduled_at
+        and data.scheduled_at != datetime.fromtimestamp(0)
+    ):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="If statsu is not schduled, do not specify schduled_at",
+        )
+
+    now = datetime.now()
+    if data.scheduled_at:
+        data_scheduled_at = data.scheduled_at.replace(tzinfo=None)
+    if (
+        data.topic_status == models.TopicStatusType.scheduled
+        and data.scheduled_at
+        and data_scheduled_at < now
+    ):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="If statsu is schduled, schduled_at must be a future time",
+        )
```

Comment:
ここで現在のstatusが scheduled の場合もまとめて判断するのでどうでしょうか
```suggestion
    now = datetime.now()
    if (
        (data.topic_status == models.TopicStatusType.scheduled or ticket.current_ticket_status.status_id == models.TopicStatusType.scheduled )
        and data.scheduled_at
        and data_scheduled_at < now
    ):
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="If statsu is schduled, schduled_at must be a future time",
        )
```

Reply:
エラーを出す条件について質問です。
今リクエストでtopic_statusがNoneかつ前回値のtopic_statusがscheduledの時に、data.scheduled_atには未来の時間が入っているかを検証する認識であっていますか？(未来の時間が入ってい無い時、エラーを出力)